src/geode/basic/uuid.cpp

```diff
@@ -68,21 +91,55 @@ namespace geode
 
     std::string uuid::string() const
     {
-        std::stringstream ss;
-        ss << std::hex << std::nouppercase << std::setfill( '0' );
+        char string[] = "00000000-0000-0000-0000-000000000000";
+        static constexpr char encode[] = "0123456789abcdef";
 
-        const uint32_t a = ( ab >> 32u );
-        const uint32_t b = ( ab & 0xFFFFFFFFu );
-        const uint32_t c = ( cd >> 32u );
-        const uint32_t d = ( cd & 0xFFFFFFFFu );
+        index_t bit = 15;
+        for( const auto i : Range{ 18 } )
+        {
+            if( i == 8 || i == 13 )
+            {
+                continue;
+            }
+            string[i] = encode[ab >> 4 * bit & 0x0f];
+            bit--;
+        }
+
+        bit = 15;
+        for( const auto i : Range{ 19, 36 } )
+        {
+            if( i == 23 )
+            {
+                continue;
+            }
+            string[i] = encode[cd >> 4 * bit & 0x0f];
+            bit--;
+        }
+
+        return string;
+    }
 
-        ss << std::setw( 8 ) << ( a ) << '-';
-        ss << std::setw( 4 ) << ( b >> 16u ) << '-';
-        ss << std::setw( 4 ) << ( b & 0xFFFFu ) << '-';
-        ss << std::setw( 4 ) << ( c >> 16u ) << '-';
-        ss << std::setw( 4 ) << ( c & 0xFFFFu );
-        ss << std::setw( 8 ) << d;
+    uuid::uuid( absl::string_view string )
+    {
+        OPENGEODE_EXCEPTION( string[8] == '-' && string[13] == '-'
```

Comment:
Verify the size of the string first, particularly to avoid access problems if string is too small ?